src/main/java/com/aws/greengrass/mqttclient/spool/Spool.java

```diff
@@ -114,10 +112,24 @@ private CloudMessageSpool setupSpooler() {
      * @throws ServiceLoadException thrown if the service cannot be located
      */
     private CloudMessageSpool getPersistenceSpoolGGService()
-            throws ServiceLoadException {
+            throws ServiceLoadException, IOException {
         GreengrassService locatedService = kernel.locate(config.getPersistenceSpoolServiceName());
         if (locatedService instanceof CloudMessageSpool) {
-            return (CloudMessageSpool) locatedService;
+            CloudMessageSpool persistenceSpool = (CloudMessageSpool) locatedService;
+            try {
+                persistentQueueSync(persistenceSpool.getAllMessageIds(), persistenceSpool);
+            } catch (SpoolerStoreException e) {
+                logger.atWarn()
+                        .kv(PERSISTENCE_SPOOL_SERVICE_NAME_KEY, config.getPersistenceSpoolServiceName())
+                        .cause(e).log("Persistence spool queue sync was not completed");
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                logger.atWarn()
+                        .kv(PERSISTENCE_SPOOL_SERVICE_NAME_KEY, config.getPersistenceSpoolServiceName())
+                        .cause(e).log("Persistence spool queue sync was not completed");
+            }
+            logger.atInfo().log("Persistent Spooler has been set up");
```

Comment:
don't log this here because it isn't true. Log this on L121.

Reply:
I was hoping we should be okay if QueueSync is stopped due to storage limitation, and still return Persistent Spooler as a spooler. Also, the case for QueueSync being stopped due to some interrupt. The function will only throw IOException if in case `getAllMessageIds` fail and in that case we default to InMemory spooler

Reply:
Let's say that you cannot setup the database. Is the persistent spooler correctly setup?

Reply:
If DB setup fails, locate would give ServiceLoadException and we will default to InMemorySpooler.

Reply:
initializing in the constructor isn't a good idea. Constructors in java have limitations compared to regular methods.

Reply:
it feels confusing to log both success and error at the same time, can we keep them separate?  For error scenario, i think we can provide more context to the customer. e.g. will this drop data? what will happen moving forward (still using persistence?)

Reply:
Sure, can I make a note of this and add it separately. I'm thinking something like "continuing with Persistent Spooler anyways" in both the catch cases so that we have some context.